lib/src/sha256.dart

```diff
@@ -105,14 +113,14 @@ class _Sha256Sink extends HashSink {
     }
 
     // Shuffle around the bits.
-    var a = digest[0];
-    var b = digest[1];
-    var c = digest[2];
-    var d = digest[3];
-    var e = digest[4];
-    var f = digest[5];
-    var g = digest[6];
-    var h = digest[7];
+    var a = _digest[0];
+    var b = _digest[1];
+    var c = _digest[2];
+    var d = _digest[3];
+    var e = _digest[4];
+    var f = _digest[5];
+    var g = _digest[6];
+    var h = _digest[7];
 
     for (var i = 0; i < 64; i++) {
       var temp1 = add32(add32(h, _bsig1(e)),
```

Comment:
On a 64-bit JS VM, it's likely that signed 32-bit numbers are efficient. Unsigned are not (but for addition and multiplication, you can use signed values without changing the outcome mod 2^32).
On a 32-bit JS VM, even 32 bit integers may be heap allocated, and only signed 31-bit integers get the full performance from "Smi" optimization.

Consider using configurable imports to switch between different implementations. It might reduce code size unless the compiler can efficiently tree-shake the unused branch.

Reply:
I've added the conditional import for vm/js implementations. Are you suggesting fixednum for the js version?

Reply:
I think fixednum *might* be more efficient, but it depends a lot on details. Maybe three smi multiplications are more efficient than two heap-number multiplications. Maybe they're not. Maybe it's sufficient to use Int32 values instead of Uint32 values for JS on 64-bit platforms. 

Maybe just commit something that works, and then we can optimize it later :)

Reply:
> Maybe just commit something that works, and then we can optimize it later :)

Indeed, make sure the tests are right, and that we have plenty of test vectors, I would rather have a few too many than later be optimizing an incorrect implementation :) hehe